segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkStorageManager.java

```diff
@@ -0,0 +1,1325 @@
+    }
+
+    /**
+     * Delete the garbage chunks.
+     * @param chunksTodelete List of chunks to delete.
+     */
+    private void collectGarbage(Collection<String> chunksTodelete) {
+        for (val chunkTodelete : chunksTodelete) {
+            try {
+                if (chunkStorage.exists(chunkTodelete)) {
+                    chunkStorage.delete(chunkStorage.openWrite(chunkTodelete));
+                    log.debug("collectGarbage - chunk={}.", chunkTodelete);
+                }
+            } catch (ChunkNotFoundException e) {
+                log.debug("Could not delete garbage chunk {}", chunkTodelete);
+            } catch (Exception e) {
+                log.warn("Could not delete garbage chunk {}", chunkTodelete);
+            }
+        }
+    }
+
+    /**
+     * Seals a StreamSegment. No further modifications are allowed on the StreamSegment after this operation completes.
+     *
+     * @param handle  A read-write SegmentHandle that points to a Segment to Seal.
+     * @param timeout Timeout for the operation.
+     * @return A CompletableFuture that, when completed, will indicate that the operation completed. If the operation
+     * failed, it will contain the cause of the failure. Notable exceptions:
+     * <ul>
+     * <li> StreamSegmentNotExistsException: When the given Segment does not exist in Storage.
+     * <li> StorageNotPrimaryException: When this Storage instance is no longer primary for this Segment (it was fenced out).
+     * </ul>
+     * @throws IllegalArgumentException If handle is read-only.
+     */
+    @Override
+    public CompletableFuture<Void> seal(SegmentHandle handle, Duration timeout) {
+        checkInitialized();
+        return execute( () -> {
+            long traceId = LoggerHelpers.traceEnter(log, "seal", handle);
+            Preconditions.checkNotNull(handle, "handle");
+            String streamSegmentName = handle.getSegmentName();
+            Preconditions.checkNotNull(streamSegmentName, "streamSegmentName");
+            Preconditions.checkArgument(!handle.isReadOnly(), "handle");
+
+            try (MetadataTransaction txn = metadataStore.beginTransaction()) {
+                SegmentMetadata segmentMetadata = (SegmentMetadata) txn.get(streamSegmentName);
+                // Validate preconditions.
+                if (null == segmentMetadata || !segmentMetadata.isActive()) {
+                    throw new StreamSegmentNotExistsException(streamSegmentName);
+                }
+
+                if (segmentMetadata.getOwnerEpoch() > this.epoch) {
+                    throw new StorageNotPrimaryException(streamSegmentName);
+                }
+
+                // seal if it is not already sealed.
+                if (!segmentMetadata.isSealed()) {
+                    segmentMetadata.setSealed(true);
+                    txn.update(segmentMetadata);
+                    txn.commit();
+                }
+
+                log.debug("seal - segment={}.", handle.getSegmentName());
```

Comment:
I think that Andrei already said that, but we need to keep a consistent format for every log-related message with this new Tier 2. For instance, we see on another class of this PR:
`log.debug("Delete chunk={} latency={}.", handle.getChunkName(), elapsed.toMillis());`
and on this class
`log.debug("seal - segment={}.", handle.getSegmentName());`
Let's define a consistent format for all the logging messages related to Tier 2 operations, if possible.

Reply:
done